### PR TITLE
Rework deduction and add with_sequence

### PR DESCRIPTION
**Pull request by Trollgeir**

### CODE REVIEW

File: src/hera/task.py [hunk 1]
```diff
@@ -89,6 +92,9 @@ class Task(IO):
         function call with the given parameters. Otherwise, if parallel is False and the list of params is a list of
         lists (each list contains a series of values to pass to the function), the task will execute as a task
         group and schedule multiple function calls in parallel.
+    with_sequence: Optional[dict] = None,
```
Comment (flaviuvadan) [inline src/hera/task.py:95]: I like this! It's super flexible. I wonder if there are some problems that some "magic" object, `Sequence`, would solve for the user?

**Comment by flaviuvadan** [comment]
Should we merge this in to `hera/4.0.0rc2` and release soon?

File: src/hera/task.py (hunk 1 above)
Comment (Trollgeir) [reply src/hera/task.py:95]: I could be ok with this, but then I feel like we should also add `value_from` as a dedicated object.

**Comment by Trollgeir** [comment]
> Should we merge this in to `hera/4.0.0rc2` and release soon?

Feel free to merge once approved :)

**Review by flaviuvadan — APPROVED** [review]
🚀

File: src/hera/task.py (hunk 1 above)
Comment (flaviuvadan) [reply src/hera/task.py:95]: Sounds reasonable. I think we can always add such magic and support `Union[dict, MagicSequence]` :)

**Pull request merged**